src/channel_state.ts

```diff
@@ -536,11 +539,41 @@ export class ChannelState<
 
         // @ts-expect-error - ImmutableArray.set exists in the documentation but not in the DefinitelyTyped types
         this.messages = messages.set(i, messageWithReaction);
+        this._syncReactions(message, messageWithReaction);
         break;
       }
     }
   }
 
+  _syncReactions(
+    eventMessage: MessageResponse<
+      AttachmentType,
+      ChannelType,
+      CommandType,
+      MessageType,
+      ReactionType,
+      UserType
+    >,
+    stateMessage: Immutable.Immutable<
```

Comment:
https://github.com/GetStream/stream-chat-js/pull/602 getting rid of this Immutable stuff here